executor/executor.go

```diff
@@ -294,6 +320,13 @@ func Run(path string, env []string, emitter screwdriver.Emitter, build screwdriv
 	}
 
 	teardownCommands := append(userTeardownCommands, sdTeardownCommands...)
+	cleanup := false
+
+	// Previous user steps ran successfully and there is no teardown step to run, clean up
+	if len(teardownCommands) == 0 && firstError == nil {
```

Comment:
Is this only for unit test? And now unit tests write to seperate file, we probably won’t need this logic any more

Reply:
No, this logic is for cleaning up for cases where all steps passed and there is no teardown step. For example:
```
- ls 
- echo hello
```
In this case, it will not go to `doRunTeardown` and no cleanup will be called. So we need this here for cleaning up.

Reply:
kk got it